app/src/main/java/com/soen490chrysalis/papilio/view/UserProfileActivity.kt

```diff
@@ -21,12 +43,20 @@ class UserProfileActivity : AppCompatActivity()
         val view = binding.root
         setContentView(view)
 
+        val userRepository : IUserRepository =
+            UserRepository(FirebaseAuth.getInstance(), userService = UserApi.retrofitService)
```

Comment:
You should never instantiate a repository in a view since it is a breach in our architecture. You should instead create a view model where we pass a factory where the repository is created.